chains/evm/contracts/offRamp/OffRampOverSuperchainInterop.sol

```diff
@@ -0,0 +1,217 @@
+
+  constructor(
+    StaticConfig memory staticConfig,
+    DynamicConfig memory dynamicConfig,
+    SourceChainConfigArgs[] memory sourceChainConfigs,
+    address crossL2Inbox,
+    ChainSelectorToChainIdConfigArgs[] memory chainSelectorToChainIdConfigArgs
+  ) OffRamp(staticConfig, dynamicConfig, sourceChainConfigs) {
+    i_crossL2Inbox = ICrossL2Inbox(crossL2Inbox);
+
+    _applyChainSelectorToChainIdConfigUpdates(new uint64[](0), chainSelectorToChainIdConfigArgs);
+  }
+
+  /// @notice Construct the Identifier from the proofData, then log hash from the message.
+  /// @param message The message to construct proofData and log hash for.
+  /// @param proofs Identifier split into an ordered array of 5 32-byte words.
+  /// @return identifier The identifier constructed from the proofs.
+  /// @return logHash The log hash constructed from the message.
+  function _constructProofs(
+    Internal.Any2EVMRampMessage memory message,
+    bytes32[] memory proofs
+  ) internal pure returns (Identifier memory identifier, bytes32 logHash) {
+    if (proofs.length != 5) revert InvalidProofsWordLength(proofs.length, 5);
+
+    identifier = Identifier({
+      origin: address(uint160(uint256(proofs[0]))),
+      blockNumber: uint256(proofs[1]),
+      logIndex: uint256(proofs[2]),
+      timestamp: uint256(proofs[3]),
+      chainId: uint256(proofs[4])
+    });
+
+    // Concatenation as opposed to abi.encode all fields is necessary to construct correct log data
+    // [0x00] event_selector
+    // [0x20] dest_chain_selector
+    // [0x40] sequence_number
+    // [0x60] offset_to_message ← should be 0x20, but would be 0x80 if abi.encode all fields at once
+    // [0x80] message...
+    logHash = keccak256(
+      bytes.concat(
+        SuperchainInterop.SENT_MESSAGE_LOG_SELECTOR,
+        bytes32(uint256(message.header.destChainSelector)),
+        bytes32(uint256(message.header.sequenceNumber)),
+        abi.encode(message)
+      )
+    );
+
+    return (identifier, logHash);
+  }
+
+  /// @notice Verify the message was indeed sent on the source chain by checking against the CrossL2Inbox.
+  /// @dev No trust assumption can be placed on the report, every field of the report can be forged.
+  /// In addition to CrossL2Inbox validation, many checks are necessary,
+  /// e.g. Identifier.chainId must match sourceChainSelector.
+  /// @param sourceChainSelector The source chain selector of the message.
+  /// @param report The execution report to verify.
+  /// @return timestampCommitted The source timestamp of the message.
+  /// @return hashedLeaves Array of 1 hashed message.
+  function _verifyMessage(
+    uint64 sourceChainSelector,
+    Internal.ExecutionReport memory report
+  ) internal virtual override returns (uint256 timestampCommitted, bytes32[] memory hashedLeaves) {
+    if (report.messages.length != 1) revert InvalidMessageCountInReport(report.messages.length, 1);
```

Comment:
Since 1 is hard coded, there should be no need to revert with `1`. Can rename the event to say "MultipleMessagesPerReportNotAllowed" or something and you don't even need any args imo